advEntryDetail.go

```diff
@@ -0,0 +1,320 @@
+	}
+	if err := ed.isAlphanumeric(ed.DFIAccountNumber); err != nil {
+		return &FieldError{FieldName: "DFIAccountNumber", Value: ed.DFIAccountNumber, Msg: err.Error()}
+	}
+	if err := ed.isAlphanumeric(ed.AdviceRoutingNumber); err != nil {
+		return &FieldError{FieldName: "AdviceRoutingNumber", Value: ed.AdviceRoutingNumber, Msg: err.Error()}
+	}
+	if err := ed.isAlphanumeric(ed.IndividualName); err != nil {
+		return &FieldError{FieldName: "IndividualName", Value: ed.IndividualName, Msg: err.Error()}
+	}
+	if err := ed.isAlphanumeric(ed.DiscretionaryData); err != nil {
+		return &FieldError{FieldName: "DiscretionaryData", Value: ed.DiscretionaryData, Msg: err.Error()}
+	}
+	if err := ed.isAlphanumeric(ed.ACHOperatorRoutingNumber); err != nil {
+		return &FieldError{FieldName: "ACHOperatorRoutingNumber", Value: ed.ACHOperatorRoutingNumber, Msg: err.Error()}
+	}
+	calculated := ed.CalculateCheckDigit(ed.RDFIIdentificationField())
+
+	edCheckDigit, _ := strconv.Atoi(ed.CheckDigit)
+
+	if calculated != edCheckDigit {
+		msg := fmt.Sprintf(msgValidCheckDigit, calculated)
+		return &FieldError{FieldName: "RDFIIdentification", Value: ed.CheckDigit, Msg: msg}
+	}
+	return nil
+}
+
+// fieldInclusion validate mandatory fields are not default values. If fields are
+// invalid the ACH transfer will be returned.
+func (ed *ADVEntryDetail) fieldInclusion() error {
+	if ed.recordType == "" {
+		return &FieldError{
+			FieldName: "recordType",
+			Value:     ed.recordType,
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.TransactionCode == 0 {
+		return &FieldError{
+			FieldName: "TransactionCode",
+			Value:     strconv.Itoa(ed.TransactionCode),
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.RDFIIdentification == "" {
+		return &FieldError{
+			FieldName: "RDFIIdentification",
+			Value:     ed.RDFIIdentificationField(),
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.DFIAccountNumber == "" {
+		return &FieldError{
+			FieldName: "DFIAccountNumber",
+			Value:     ed.DFIAccountNumber,
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.AdviceRoutingNumber == "" {
+		return &FieldError{
+			FieldName: "AdviceRoutingNumber",
+			Value:     ed.AdviceRoutingNumber,
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.IndividualName == "" {
+		return &FieldError{
+			FieldName: "IndividualName",
+			Value:     ed.IndividualName,
+			Msg:       msgFieldRequired + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.ACHOperatorRoutingNumber == "" {
+		return &FieldError{
+			FieldName: "ACHOperatorRoutingNumber",
+			Value:     ed.ACHOperatorRoutingNumber,
+			Msg:       msgFieldInclusion + ", did you use NewADVEntryDetail()?",
+		}
+	}
+	if ed.JulianDateDay == 0 {
```

Comment:
```suggestion
	if ed.JulianDateDay <= 0 {
```

Are Julian Dates better represented by a uint?

Reply:
yeah it should be unsigned.  good point